### DISCUSSION TITLE
Migration to 0.15.1 ?

### DISCUSSION THREAD

**Discussion by Quafadas** [disneystreaming/smithy4s#386] [Q&A] [answered]
I've had a best effort at reading the docs, and I'm unclear if I need to change something, or whether I've stumbled on a bug. 

build.sc
```
val smithy4sVersion = "0.15.1"
```
ClaimNotificationService.smithy
```
namespace ils.claims

use smithy4s.api#simpleRestJson
use smithy4s.api#UUID
use smithy4s.api#uuidFormat
use smithy4s.meta#packedInputs

@uuidFormat
string ClaimNotificationId
``` 
Then clean compile workspace.
This successfully generates it's code, which makes me think, that the usage of the UUID decorator is intended to remain valid. This particular example, doesn't seem to compile for me though... 

```
[error] -- [E008] Not Found Error: C:\temp\ilsclaims\out\shared\smithy4sCodegen.dest\scala\ils\claims\ClaimNotificationId.scala:13:111 
[error] 13 |  implicit val schema : smithy4s.Schema[ClaimNotificationId] = bijection(underlyingSchema, ClaimNotificationId.make, (_ : ClaimNotificationId).value)    
[error]    |                                                                                           ^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |value make is not a member of object ils.claims.ClaimNotificationId - did you mean ClaimNotificationId.value?
[error] -- [E008] Not Found Error: C:\temp\ilsclaims\out\shared\smithy4sCodegen.dest\scala\ils\claims\ClaimReportId.scala:13:99 
[error] 13 |  implicit val schema : smithy4s.Schema[ClaimReportId] = bijection(underlyingSchema, ClaimReportId.make, (_ : ClaimReportId).value)
[error]    |                                                                                     ^^^^^^^^^^^^^^^^^^
```
The generated code appears to be this; 
```
package ils.claims

import java.util.UUID
import smithy4s.Newtype
import smithy4s.schema.Schema._

object ClaimNotificationId extends Newtype[UUID] {
  val id: smithy4s.ShapeId = smithy4s.ShapeId("ils.claims", "ClaimNotificationId")
  val hints : smithy4s.Hints = smithy4s.Hints(
    smithy4s.api.UuidFormat(),
  )
  val underlyingSchema : smithy4s.Schema[UUID] = uuid.withId(id).addHints(hints)
  implicit val schema : smithy4s.Schema[ClaimNotificationId] = bijection(underlyingSchema, ClaimNotificationId.make, (_ : ClaimNotificationId).value)
}
```
Would this be expected?

**Comment by Baccata** [comment, marked as answer]
@Quafadas have you tried `mill clean` and recompile ? If you did try and it didn't work, any chance you could share your full build ? 

Also, when you get things like this, feel free to open an Issue instead of a discussion

**Reply by Baccata** [reply]
> Then clean compile workspace.

Oh, you did, sorry. In theory `.make` should not be regenerated anymore, so something's obviously not picked up your change in config. When you say `clean compile workspace`, what do you do exactly ?

**Reply by Baccata** [reply]
It occured to me that you probably have the following at the top of your build file. 

```
import $ivy.`com.disneystreaming.smithy4s:smithy4s-codegen:version
```

at the top of the file, and you might have forgotten to change it. If that's the case, then you'll be able to change your `val smithy4sVersion = "0.15.1"`  for `val smithy4sVersion = smithy4s.codegen.BuildInfo.version` inside your build file in order to prevent the build-time dependency and the runtime-dependency from falling out of sync with one another.

**Reply by Quafadas** [reply]
Urgh. it is exactly this. 
```
import $ivy.`com.disneystreaming.smithy4s:smithy4s-codegen:version
```
I've followed your suggestion and imported the smithy version directly from that import. I have to admit, I wasn't aware one could do that. One less thing to keep in sync. Thankyou again... and apologies for the spam

**Reply by daddykotex** [reply]
It's no spam @Quafadas . You have been tremendously helpful.

You are enthusiastic and grateful for the project and you keep us on our toes! That's a good thing. You makes us see all kinds of things we overlook because we are too close to the project and often don't use it in the same scenario as you.

Please keep asking any questions you might have.
